### ISSUE TITLE
Warn about creating a setter or getter without defining a property

### ISSUE THREAD

**Issue by omarandlorraine** [PyCQA/pylint#7172]
### Bug description

Here are a couple of classes for holding and retrieving a value.

It demonstrates a mistake I genuinely made (not that good at Python ... yet!)

The contents of the file duff.py:

```
# pylint: disable=C0116,C0114,C0303,C0115,R0903
class Duff:
    def __init__(self):
        self._duff = None

    @property
    def duff(self):
        return self._duff

    @duff.setter
    def duff(self, value):
        self._duff = value

class DuffPlusOne(Duff):
    
    def __init__(self):
        super().__init__()
        self._duff = None

    @duff.setter
    def duff(self, value):
        self._duff = value + 1
```

### Command used

```shell
pylint -E duff.py
```

### Pylint output

```shell
Your code has been rated at 10.0/10
```

### Expected behavior

I would have expected pylint to tell me that I can't have a setter in DuffPlusOne since there's no matching getter.

### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.11.2
Python 3.9.2 (default, Feb 28 2021, 17:03:44)
[GCC 10.2.1 20210110]
```

### OS / Environment

Debian Buster

**Comment by omarandlorraine** [comment]
I want to question the "False Positive" label. The description for this label is "A message is emitted but nothing is wrong with the code". But in this case, it's actually the opposite. I wonder if there's a "False Negative" label we can use.

**Comment by Pierre-Sassoulas** [comment]
Thanks for noticing, you're right of course, my bad !

**Comment by omarandlorraine** [comment]
Is `@duff.setter` a so-called decorator? In that case, I wonder if the same issue goes for any other unknown decorator such as `@memoize` or any other name.

**Comment by DanielNoord** [comment]
Yeah it's a decorator, but of a special kind. See:
https://docs.python.org/3/library/functions.html#property.

I haven't tested, but can you have a `getter` without a `setter`? If so I think both `property-missing-getter` and `property-missing-setter` would be good additions.

**Comment by mbyrnepr2** [comment]
Example can be simplified to:
```python
class Duff:
    def __init__(self):
        self._duff = None

    @duff.setter
    def duff(self, value):
        self._duff = value
```

```python
  File "/Users/markbyrne/programming/x.py", line 1, in <module>
    class Duff:
  File "/Users/markbyrne/programming/x.py", line 5, in Duff
    @duff.setter
NameError: name 'duff' is not defined
```

**Comment by DanielNoord** [comment]
Sorry, I should have investigated this.
The issue is as follows, with messages to be raised in the comments:

```python
class Fruit:
    def __init__(self):
        self._taste = None

    @property
    def taste(self):
        return self._taste


class Apple(Fruit):
    def __init__(self):
        self._taste = None

    @taste.setter  # [setter-without-property]
    def taste(self, value):
        self._taste = value


class Banana(Fruit):
    def __init__(self):
        self._taste = None

    @taste.getter  # [getter-without-property]
    def taste(self):
        return self._taste
```

The issue is that you can't "inherit" the property from the base class to have a setter or getter specific to the child class. You need to redefine the property in the child class.

**Comment by omarandlorraine** [comment]
> I think having a property without the setter is acceptable Python (creates a read-only property):

Does this point vary between Python2 and Python3?

**Comment by mbyrnepr2** [comment]
Generally speaking (irrespective of properties/getters/setters) this is what Pylint is not picking up on:
```python
class MyClass:
    @hello
    def hello(self):
        ...
```

python example.py
```python
Traceback (most recent call last):
  File "/Users/markbyrne/programming/example.py", line 1, in <module>
    class MyClass:
  File "/Users/markbyrne/programming/example.py", line 2, in MyClass
    @hello
NameError: name 'hello' is not defined. Did you mean: 'help'?
```
